### PR TITLE
Added Q&A apis and auth api changes

### PR DESCRIPTION
**Pull request by pallavichavda04**

### CODE REVIEW

File: src/api/QA/qa.service.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    async addAnswer(createAnswerDto: CreateAnswerDto, userData: any): Promise<ResponseDto> {
+        try {
+            const question = await this.questionModel.findById(createAnswerDto.questionId);
+            if (!question) {
+                return { statusCode: HttpStatus.NOT_FOUND, message: MESSAGE.QUESTION_NOT_FOUND }
+            }
+
+            if (question.userId !== userData.userId) {
+                return { statusCode: HttpStatus.UNAUTHORIZED, message: MESSAGE.INVALID_CREDENTIALS }
```
Comment (yuvraj-bytes) [inline src/api/QA/qa.service.ts:48]: change message

File: src/api/QA/qa.service.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            }
+
```
Comment (yuvraj-bytes) [inline src/api/QA/qa.service.ts:50]: Add Condition for only owner of course can reply.

File: src/api/QA/qa.service.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            question.answers.push({
+                userId: userData.userId,
+                answers: createAnswerDto.answer,
```
Comment (yuvraj-bytes) [inline src/api/QA/qa.service.ts:53]: Fix this.

**Comment by admsev** [comment]
@gooroodev please review

**Comment by gooroodev** [comment]
Thanks, @admsev, I’m ready to assist!

### 1. Summary of Changes

The pull request introduces the following changes:

1. **Q&A APIs**:
    - Added DTOs for creating questions and answers (`create-question.dto.ts`, `create-answer.dto.ts`).
    - Implemented a controller (`qa.controller.ts`) for handling Q&A endpoints.
    - Created a service (`qa.service.ts`) to manage the business logic for Q&A.
    - Defined Mongoose schemas for questions and answers (`question.schema.ts`, `answer.schema.ts`).
    - Integrated the Q&A module (`qa.module.ts`) into the application.
    - Updated constants with new messages and notifications related to Q&A.

2. **Auth API Changes**:
    - Modified the password reset functionality in `auth.service.ts` to directly set the new password without hashing.

3. **General**:
    - Updated `app.module.ts` to include the new Q&A module.
    - Minor changes in user schema (`user.schema.ts`).

### 2. Issues, Bugs, or Typos

#### 2.1. Password Reset in `auth.service.ts`
- **Issue**: The password is set directly without hashing, which is a security vulnerability.
- **Proposed Fix**: Reintroduce hashing for the new password.

```typescript
user.password = await bcrypt.hash(resetPasswordDto.newPassword, 10);
```

#### 2.2. Typo in Notification Schema Import Path
- **Issue**: Typo in the import path for the notification schema in `qa.module.ts`.
- **Proposed Fix**: Correct the import path.

```typescript
import { Notification, NotificationSchema } from '../notification/schema/notification.schema';
```

#### 2.3. Missing Newline at End of File
- **Issue**: Missing newline at the end of `answer.schema.ts`.
- **Proposed Fix**: Add a newline at the end of the file.

```typescript
export const AnswerSchema = SchemaFactory.createForClass(Answer);
```

#### 2.4. Unused Imports in `user.schema.ts`
- **Issue**: Unused imports in `user.schema.ts`.
- **Proposed Fix**: Remove the unused import.

```typescript
import { Document } from 'mongoose'; // Remove this line if it's not used
```

### 3. General Review of Code Quality and Style

#### 3.1. Code Quality
- **Modularity**: The code is well-modularized, with clear separation between DTOs, controllers, services, and schemas.
- **Error Handling**: The service methods include try-catch blocks for error handling, which is good practice.
- **Validation**: DTOs use class-validator for input validation, ensuring data integrity.

#### 3.2. Code Style
- **Consistency**: The code follows a consistent style in terms of indentation, spacing, and naming conventions.
- **Comments**: The code lacks comments, which could be helpful for understanding the logic, especially in complex methods.
- **Type Safety**: TypeScript's type safety features are used effectively, enhancing code reliability.

#### 3.3. Suggestions for Improvement
- **Comments and Documentation**: Adding comments and method documentation would improve code readability and maintainability.
- **Unit Tests**: Implement unit tests for the new Q&A functionalities to ensure they work as expected and to facilitate future refactoring.

### Conclusion
Overall, the pull request introduces valuable Q&A functionalities and makes necessary changes to the authentication service. Addressing the identified issues and considering the suggestions for improvement will enhance the code's security, readability, and maintainability.

Yours, [Gooroo.dev](https://github.com/apps/gooroo-dev). React or reply to share your feedback with me!

**Pull request merged**